frappe/public/js/frappe/views/reports/query_report.js

```diff
@@ -135,6 +137,47 @@ frappe.views.QueryReport = Class.extend({
 			frappe.render_grid({grid:this.grid, title:__(this.report_name)});
 		}
 	},
+	pdf_report: function() {
+		if(this.html_format) {
+			var content = frappe.render(this.html_format,
+				{data: this.dataView.getItems(), filters:this.get_values(), report:this});
+
+			//Render Report in HTML
+			var html = frappe.render_template("print_template", {content:content, title:__(this.report_name)});
+		} else {
+			var columns = this.grid.getColumns();
+			var data = this.grid.getData().getItems();
+			var content = frappe.render_template("print_grid", {columns:columns, data:data, title:__(this.report_name)})
+
+			//Render Report in HTML
+			var html = frappe.render_template("print_template", {content:content, title:__(this.report_name)});
+		}
+
+		//Create a form to place the HTML content 
+		var formData = new FormData();
+
+		//Push the HTML content into an element
+		formData.append("html", html);
+		var blob = new Blob([], { type: "text/xml"});
+		//formData.append("webmasterfile", blob);
+		formData.append("blob", blob);
+
+		var xhr = new XMLHttpRequest();
+		xhr.open("POST", '/api/method/frappe.templates.pages.print.report_to_pdf');
+		xhr.setRequestHeader("X-Frappe-CSRF-Token", frappe.csrf_token);
+		xhr.responseType = "arraybuffer";
+
```

Comment:
Dont think its possible, I need to return an arraybuffer to generate the PDF on the fly. Perhaps you have an idea of how we can use frappe.call?